api/client_test.go

```diff
@@ -1,71 +1,92 @@
+	t.Cleanup(func() {
+		_ = os.Unsetenv(EnvAddress)
+		_ = os.Unsetenv(EnvTLSCACert)
+		_ = os.Unsetenv(EnvTLSCAPath)
+		_ = os.Unsetenv(EnvTLSClientCert)
+		_ = os.Unsetenv(EnvTLSClientKey)
+		_ = os.Unsetenv(EnvTLSSSLVerify)
+	})
+
 	url := "https://1.2.3.4:5678"
 	caCert := "test/path/ca.pem"
 	caPath := "test/path"
 	clientCert := "test/path/client.pem"
 	clientKey := "test/path/key.pem"
 	sslVerify := "false"
 
-	err := os.Setenv(EnvAddress, url)
-	require.NoError(t, err)
-	defer os.Setenv(EnvAddress, "")
-
-	err = os.Setenv(EnvTLSCACert, caCert)
-	require.NoError(t, err)
-	defer os.Setenv(EnvTLSCACert, "")
-
-	err = os.Setenv(EnvTLSCAPath, caPath)
-	require.NoError(t, err)
-	defer os.Setenv(EnvTLSCAPath, "")
-
-	err = os.Setenv(EnvTLSClientCert, clientCert)
-	require.NoError(t, err)
-	defer os.Setenv(EnvTLSClientCert, "")
-
-	err = os.Setenv(EnvTLSClientKey, clientKey)
-	require.NoError(t, err)
-	defer os.Setenv(EnvTLSClientKey, "")
-
-	os.Setenv(EnvTLSSSLVerify, sslVerify)
-	require.NoError(t, err)
-	defer os.Setenv(EnvTLSSSLVerify, "")
+	require.NoError(t, os.Setenv(EnvAddress, url))
+	require.NoError(t, os.Setenv(EnvTLSCACert, caCert))
+	require.NoError(t, os.Setenv(EnvTLSCAPath, caPath))
+	require.NoError(t, os.Setenv(EnvTLSClientCert, clientCert))
+	require.NoError(t, os.Setenv(EnvTLSClientKey, clientKey))
+	require.NoError(t, os.Setenv(EnvTLSSSLVerify, sslVerify))
 
-	config := DefaultClientConfig()
+	clientConfig := DefaultClientConfig()
 
-	assert.Equal(t, url, config.Addr)
-	assert.Equal(t, caCert, config.TLSConfig.CACert)
-	assert.Equal(t, caPath, config.TLSConfig.CAPath)
-	assert.Equal(t, clientCert, config.TLSConfig.ClientCert)
-	assert.Equal(t, clientKey, config.TLSConfig.ClientKey)
+	require.Equal(t, url, clientConfig.Addr)
+	require.Equal(t, caCert, clientConfig.TLSConfig.CACert)
+	require.Equal(t, caPath, clientConfig.TLSConfig.CAPath)
+	require.Equal(t, clientCert, clientConfig.TLSConfig.ClientCert)
+	require.Equal(t, clientKey, clientConfig.TLSConfig.ClientKey)
 
 	expectedSSLVerify, err := strconv.ParseBool(sslVerify)
 	require.NoError(t, err)
-	assert.Equal(t, expectedSSLVerify, config.TLSConfig.SSLVerify)
+	require.Equal(t, expectedSSLVerify, clientConfig.TLSConfig.SSLVerify)
 }
 
-func TestDefaultClientConfig_OnlyDefaults(t *testing.T) {
+func Test_DefaultClientConfig_Defaults(t *testing.T) {
 	caCert := ""
 	caPath := ""
 	clientCert := ""
 	clientKey := ""
 
-	config := DefaultClientConfig()
+	clientConfig := DefaultClientConfig()
+
+	require.Equal(t, DefaultAddress, clientConfig.Addr)
+	require.Equal(t, caCert, clientConfig.TLSConfig.CACert)
+	require.Equal(t, caPath, clientConfig.TLSConfig.CAPath)
+	require.Equal(t, clientCert, clientConfig.TLSConfig.ClientCert)
+	require.Equal(t, clientKey, clientConfig.TLSConfig.ClientKey)
+	require.Equal(t, DefaultSSLVerify, clientConfig.TLSConfig.SSLVerify)
+}
+
+func Test_ClientPort(t *testing.T) {
+	expectedPort := rand.Intn(10000)
```

Comment:
Looks like this test is using random data, which I haven't seen in other tests in this file. Do you mind sharing context behind using a random vs. hard-coded value?

Reply:
There isn't a good reason here for using a random value really. I will switch it to a constant to simplify it.

Reply:
Addressed in https://github.com/hashicorp/consul-terraform-sync/pull/663/commits/63d9a6d23abf5cec7b5928f73d6f56da5836fbcb